examples/set-image-advanced/.expected/results.yaml

```diff
@@ -0,0 +1,10 @@
+apiVersion: kpt.dev/v1
+kind: FunctionResultList
+metadata:
+  name: fnresults
+exitCode: 0
+items:
+  - image: gcr.io/kpt-fn/set-image:unstable
+    exitCode: 0
+    results:
+      - message: set image nginx:1.20.2 to bitnami/nginx:1.21.4 for 2 field(s)
```

Comment:
We are showing a summary (# of the fields that has been updated).
Should we actually show each place (GVKNN + fields path) that has been changed?
Also we should set severity to info.

Reply:
The main reason I opted for this UX is that there is a limitation when iterating over a `SequenceNode` (e.g. `spec.containers[]`), where the `FieldPath` of any nested nodes treat the `SequenceNode` as the root.
e.g.
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: app
spec:
  containers:
  - name: store
    image: redis:6.2.6
```
The `FieldPath` of the `RNode` for `image` here would be `image` rather than something you might expect like `spec.containers[].image`. I opted to display this summary to provide a more intuitive UX given this limitation.

This is actually something that impacts `set-annotations` and `set-labels` as well, but we haven't seen this yet since no test cases involve a fieldspec including a `SequenceNode`. However, this behavior quickly arises for `set-image` since a default fieldspec includes a path with `SequenceNode`.

To change this would require some updates to the relevant behavior in kustomize - do you think we should change this behavior to display the full `FieldPath` past a `SequenceNode`?

Here is an example output:
```bash
$ kpt fn render set-image-simple
Package "set-image-simple": 
[RUNNING] "gcr.io/kpt-fn/set-image:unstable"
[PASS] "gcr.io/kpt-fn/set-image:unstable" in 1.5s
  Results:
    [info]: set image nginx:1.20.2 to bitnami/nginx:1.21.4 for 2 fields

Successfully executed 1 function(s) in 1 package(s).
```
Severity already seems to be `info`

Reply:
The behavior with SequenceNode seems to be a bug. Can you please file an issue in kyaml? If it's a bug, can you check this is a bug that can be fixed within 1 hour? If so, that'd be great if you can fix it. Otherwise, we can keep the current behavior in this PR.

Reply:
Fixed by this PR: https://github.com/kubernetes-sigs/kustomize/pull/4425

I've updated this PR to consume the upstream changes and have changed the UX to include GVK + FieldPath. Let me know what you think